File: android/data/src/main/kotlin/com/bolsillo/data/ai/LastUsedExpenseClassifier.kt
package com.bolsillo.data.ai

import com.bolsillo.domain.ai.ClassificationInput
import com.bolsillo.domain.ai.ClassificationResult
import com.bolsillo.domain.ai.ExpenseClassifier
import com.bolsillo.domain.port.TransactionRepository
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Stub classifier used until spec 003 ships the real cascade (research R7).
 * Returns the last-used category id with confidence 0.0; never blocks save
 * (Article V). `learn()` is a no-op — on-device personalization is E4.
 */
@Singleton
class LastUsedExpenseClassifier
    @Inject
    constructor(private val transactions: TransactionRepository) : ExpenseClassifier {
        override suspend fun suggest(input: ClassificationInput): ClassificationResult {
            val last = transactions.lastUsed()
            return ClassificationResult(
                topCategoryId = last?.categoryId,
                confidence = 0.0,
                alternatives = emptyList(),
            )
        }

        override suspend fun learn(
            input: ClassificationInput,
            chosenCategoryId: String,
        ) = Unit
    }

File: ios/Packages/Bolsillo/Sources/BolsilloData/AI/LastUsedExpenseClassifier.swift
import Foundation
import BolsilloDomain

public final class LastUsedExpenseClassifier: ExpenseClassifier, @unchecked Sendable {
    private let transactions: any TransactionRepository

    public static let defaultThreshold: Double = 0.75

    public init(transactions: some TransactionRepository) {
        self.transactions = transactions
    }

    public func suggest(_ input: ClassificationInput) async -> ClassificationResult {
        let last = await transactions.lastUsed()
        return ClassificationResult(
            topCategoryId: last?.categoryId,
            confidence: 0.0,      // stub: R8 — real ML is spec 003
            alternatives: []
        )
    }

    public func learn(_ input: ClassificationInput, chosenCategoryId: String) async {
        // no-op: personalization is E4 (spec 003)
    }
}
